diff -Naur gcc-7.1.0/gcc/config/m68k/asdf.h gcc-7.1.0-asdf/gcc/config/m68k/asdf.h
--- gcc-7.1.0/gcc/config/m68k/asdf.h	1970-01-01 01:00:00.000000000 +0100
+++ gcc-7.1.0-asdf/gcc/config/m68k/asdf.h	2017-07-26 23:32:55.941155733 +0200
@@ -0,0 +1,37 @@
+/* Configuration file for an m68k ASDF target. */
+
+#undef CPP_SPEC
+#define CPP_SPEC "-isystem %R/include_h"
+
+/*
+ * -fleading-underscore is essential to link with precompiled Amiga libraries
+ */
+#undef CC1_SPEC
+#define CC1_SPEC "-fleading-underscore -fomit-frame-pointer"
+
+#undef LINK_SPEC
+#define LINK_SPEC \
+" \
+-Cgnu -Bstatic -bamigahunk -S -nostdlib -sc -gc-all -L%R/linker_libs \
+%{-target-help:-h} \
+%{-help:-h} \
+"
+
+/* NOTE: -lg is same as -lc with Newlib (hard link in file system) */
+/* TODO: get rid of -lamiga dependency */
+#undef  LIB_SPEC
+#define LIB_SPEC "-lglue -lc -lamiga"
+
+#undef STARTFILE_SPEC
+#define STARTFILE_SPEC \
+" \
+%{!qnstart: ustart.o%s} \
+%{qnstart: nstart.o%s} \
+"
+
+#undef ENDFILE_SPEC
+#define ENDFILE_SPEC ""
+
+#undef LINKER_NAME
+#define LINKER_NAME "asdf-vlink"
+
diff -Naur gcc-7.1.0/gcc/config/m68k/asdf.opt gcc-7.1.0-asdf/gcc/config/m68k/asdf.opt
--- gcc-7.1.0/gcc/config/m68k/asdf.opt	1970-01-01 01:00:00.000000000 +0100
+++ gcc-7.1.0-asdf/gcc/config/m68k/asdf.opt	2017-07-26 23:32:55.941155733 +0200
@@ -0,0 +1,7 @@
+; m68k ASDF options
+
+qnstart
+Driver Target Report
+Link with nano startup code (limited support for C standard library).
+
+; This comment is to ensure we retain the blank line above.
diff -Naur gcc-7.1.0/gcc/config/m68k/t-asdf gcc-7.1.0-asdf/gcc/config/m68k/t-asdf
--- gcc-7.1.0/gcc/config/m68k/t-asdf	1970-01-01 01:00:00.000000000 +0100
+++ gcc-7.1.0-asdf/gcc/config/m68k/t-asdf	2017-07-26 23:32:55.941155733 +0200
@@ -0,0 +1,4 @@
+# Amiga computers have 680x0 CPUs
+
+M68K_MLIB_CPU += && (match(MLIB, "^680"))
+M68K_ARCH := m68k
diff -Naur gcc-7.1.0/gcc/config.gcc gcc-7.1.0-asdf/gcc/config.gcc
--- gcc-7.1.0/gcc/config.gcc	2017-03-24 14:59:51.000000000 +0100
+++ gcc-7.1.0-asdf/gcc/config.gcc	2017-07-26 23:32:40.097210823 +0200
@@ -1911,6 +1911,16 @@
 	tm_file="${tm_file} m68k/m68k-none.h m68k/m68kelf.h dbxelf.h elfos.h newlib-stdint.h m68k/m68kemb.h m68k/m68020-elf.h"
 	tm_defines="${tm_defines} MOTOROLA=1"
 	tmake_file="m68k/t-floatlib m68k/t-m68kbare m68k/t-m68kelf"
+	case ${target} in
+	m68k-asdf-elf*)
+		tm_file="${tm_file} m68k/asdf.h"
+		# NOTE: m68k/t-asdf must appear before m68k/t-mlibs
+		tmake_file="$tmake_file m68k/t-asdf"
+		extra_options="${extra_options} m68k/asdf.opt"
+		;;
+	*)
+		;;
+	esac
 	# Add multilibs for targets other than fido.
 	case ${target} in
 	fido-*-elf*)
diff -Naur gcc-7.1.0/gcc/testsuite/gcc.c-torture/execute/builtins/builtins.exp gcc-7.1.0-asdf/gcc/testsuite/gcc.c-torture/execute/builtins/builtins.exp
--- gcc-7.1.0/gcc/testsuite/gcc.c-torture/execute/builtins/builtins.exp	2017-01-01 13:07:43.000000000 +0100
+++ gcc-7.1.0-asdf/gcc/testsuite/gcc.c-torture/execute/builtins/builtins.exp	2017-07-26 23:33:28.725081017 +0200
@@ -42,7 +42,7 @@
    lappend additional_flags "-Wl,-multiply_defined,suppress"
 }
 if { [istarget *-*-eabi*]
-     || [istarget *-*-elf]
+     || ([istarget *-*-elf] && ![istarget m68k-asdf-elf])
      || [istarget *-*-mingw*]
      || [istarget *-*-rtems*] } {
    lappend additional_flags "-Wl,--allow-multiple-definition"
